docs/proposals/002-Versioning_Proposal.md

```diff
@@ -0,0 +1,209 @@
+
+For non-latest versions, I will implement a notification banner:
+
+- Subtle indicator that appears in an unobtrusive location
+- Brief message indicating a newer version exists
+- Link to the equivalent page in the latest version
+- Designed to complement the existing UI without disrupting the user experience
+
+
+![Banner](banner.png)
+
+This banner ensures users are immediately aware when viewing outdated content while providing a direct path to the latest version.
+
+### 5. Cross-Version Navigation System
+
+I will develop a simple navigation system between versions:
+
+- Maps common pages across versions
+- Falls back to the main page when direct equivalents aren't available
+- Uses a straightforward URL-based approach
+
+### 6. Integration with Jupyter Book
+
+I will make minimal changes to the Jupyter Book configuration to:
+
+- Load version management components
+- Integrate with the existing theme
+- Preserve current customizations
+- Support the version selector UI
+
+This will involve updating the _config.yml file to include the necessary JavaScript and CSS files while maintaining compatibility with the existing configuration.
+
+### 7. Implementation Dependencies
+
+- **Step 0** (Signed-Off Design) defines the versioning plan using Git tags
+- **Step 1** (Version Architecture) is foundational and must be implemented first
+- **Step 2** (GitHub Actions Workflow) can be developed in parallel with Step 1
+- **Steps 3-4** (UI Components) depend on Step 1
+- **Step 5** (Cross-Version Navigation) depends on Steps 1-3
+- **Step 6** (Jupyter Book Integration) integrates all previous components
+
+### Implementation Timeline
+
+**Phase 1: Foundation & Infrastructure (Weeks 1-3)**
+- Develop version data architecture
+- Configure GitHub Actions workflow
+- Create initial deployment tests
+
+**Phase 2: Core UI Development (Weeks 4-6)**
+- Implement version selector component
+- Create outdated version notification banner
+- Develop basic cross-version navigation
+
+**Phase 3: Testing & Refinement (Weeks 7-9)**
+- Enhance cross-version navigation
+- Implement special case handling
+- Conduct cross-browser testing
+- Verify accessibility compliance
+
+**Phase 4: Refinement & Documentation (Weeks 10-12)**
+- Optimize performance
+- Refine responsive behavior
+- Create technical documentation
+- Prepare user guides
+
+## Websites following a similar approach
+- https://pandas.pydata.org/docs/
+- https://scikit-learn.org/stable/  
+- https://matplotlib.org/stable/  
+
+## Alternative Approach
+
+I've evaluated several alternative approaches based on thorough analysis of documentation versioning systems, I came to the conclusion that these two approaches are best according to industry standards with minimal changes in repository structure:
+
+### Read the Docs Migration
+
+**Approach**: Migrate from GitHub Pages to Read the Docs platform.
+
+**Advantages**:
+- Built-in versioning system
+- Established version selector UI
+
+**Disadvantages**:
```

Comment:
A few others to add:

- This also requires a paid account to run ad free
- depends on a more complex service than just hosting static webpages, which can be done on a multitude of services.